src/main/io/beeper.c

```diff
@@ -403,6 +394,32 @@ void beeperUpdate(timeUs_t currentTimeUs)
 }
 
 /*
+ * Beeper motor/esc handler function to be called periodically in loop. 
+ */
+#ifdef USE_DSHOT
+void beeperMotorUpdate(timeUs_t currentTimeUs)
+{
+    static timeUs_t nextBeepTime = 0;
```

Comment:
This should really be synchronised with the normal beeper output. I assume that most users will want to use the Dshot beacon in conjunction with the beeper that they already have on their quad. Having them work in an unsynchronised fashion will be less effective (and sound worse) than having them go at the same time.

Reply:
This currently isn't working as intended.  For the BEEPER_RX_SET it is calling the ON and OFF off the beeper every 100ms it looks like.  The dshot beeper is triggering on both and on and off.  I don't think dshot beeper is keeping up to that.  Although I don't have an actual beeper on any of my quads so I can't verify that.

Reply:
Also I think the primary use case for dshot beacon is when people don't have an installed buzzer.  At the last race I don't think anyone had a buzzer installed, or wasn't enabling it.

Reply:
Yes, I had that impression too, the beacon should probably only be activated when the beeper is set to 'on', but not 'off'.

I think positioning the Dshot beacon as something to be only used on craft that do not have a beeper installed is underselling it. Outside of races, a majority of users probably do have a beeper. Having the tones synchronised will improve the effect for those users, to no detrimental effect for users without a beeper.

Reply:
#4360 fixes the problem of the Dshot beacon being triggered when the beeper is switched off.

Reply:
I don't know how quickly exactly the blheli_s can respond to new beeps.  Maybe 100-300ms? I moved the times out we always get a consistent motor beacon.  This could probably be synced with start of the beep, but it may miss some subsequent beeps.  I don't think it can respond fast enough to act exactly the same way in all cases.

Reply:
Good point. Making it configurable is probably the only way to cover all eventualities. I think there is a version of ESC firmware that uses the startup sound for one of the beacon tones. But even then, for the users that do have a beeper, having the beeping and the Dshot beacon synchronised will be preferrable (every n-th beeper beep or so).